usaspending_api/broker/management/commands/fabs_nightly_loader.py

```diff
@@ -65,8 +65,44 @@ def delete_stale_fabs(ids_to_delete=None):
         if not ids_to_delete:
             return
 
-        # This cascades deletes for TransactionFABS & Awards in addition to deleting TransactionNormalized records
-        TransactionNormalized.objects.filter(assistance_data__afa_generated_unique__in=ids_to_delete).delete()
+        transactions = TransactionNormalized.objects.filter(
+            assistance_data__afa_generated_unique__in=ids_to_delete).values_list('id')
+        transaction_ids_delete = [str(delete_result[0]) for delete_result in transactions]
+        lt_awards = Award.objects.filter(latest_transaction_id__in=transaction_ids_delete).values_list('id')
+        award_ids_delete = [str(delete_result[0]) for delete_result in lt_awards]
+        lt_trans = TransactionNormalized.objects.filter(award_id__in=award_ids_delete).values_list('id')
+        transaction_ids_delete.extend([str(delete_result[0]) for delete_result in lt_trans])
+        transaction_id_list = ','.join(set(transaction_ids_delete))
+        award_ids_list = ','.join(award_ids_delete)
+
+        db_cursor = connections['default'].cursor()
+
+        # Financial Accounts by Awards
+        fa = 'DELETE ' \
+             'FROM "financial_accounts_by_awards" fa ' \
+             'WHERE fa."award_id" IN ({});'.format(award_ids_list)
+        # Subawards
+        sub = 'DELETE ' \
+              'FROM "awards_subaward" sub ' \
+              'WHERE sub."award_id" IN ({});'.format(award_ids_list)
+        # Transaction FPDS
+        fpds = 'DELETE ' \
+               'FROM "transaction_fpds" tf ' \
+               'WHERE tf."transaction_id" IN ({});'.format(transaction_id_list)
+        # Transaction FABS
+        fabs = 'DELETE ' \
+               'FROM "transaction_fabs" tf ' \
+               'WHERE tf."transaction_id" IN ({});'.format(transaction_id_list)
+        # Transaction Normalized
+        tn = 'DELETE ' \
+             'FROM "transaction_normalized" tn ' \
+             'WHERE tn."id" IN ({});'.format(transaction_id_list)
+        # Awards
+        awards = 'DELETE ' \
```

Comment:
Would we want to delete the award every time? or just when the _only_ transaction for that award is deleted?

Reply:
If I understand the original logic, using Django the script would delete the award, then recreate it later if (and only if) there is a new transaction which belongs with that award.

I guess we really only want to delete the award if there are no other transactions associated, if not would need to update the award record instead of deleting. But does that fall within the scope of getting the nightly loader to work with the minor changes to the matview models?